src/MiniCover/Instrumentation/CustomAssemblyResolver.cs

```diff
@@ -1,19 +1,40 @@
-﻿using MiniCover.Utils;
+﻿using Microsoft.Extensions.DependencyModel;
+using MiniCover.Utils;
 using Mono.Cecil;
 using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace MiniCover.Instrumentation
 {
     public class CustomAssemblyResolver : DefaultAssemblyResolver
     {
-        public CustomAssemblyResolver() : base()
+        private DependencyContext _dependencyContext;
+
+        public CustomAssemblyResolver(string assemblyDirectory)
         {
             RemoveSearchDirectory(".");
             RemoveSearchDirectory("bin");
+
+            AddSearchDirectory(assemblyDirectory);
+
+            _dependencyContext = DepsJsonUtils.LoadDependencyContext(assemblyDirectory);
+
+            var runtimeConfigPath = Directory.GetFiles(assemblyDirectory, "*.runtimeconfig.dev.json", SearchOption.TopDirectoryOnly)
```

Comment:
I moved the runtimeConfig logic that you wrote to the resolver.

Reply:
btw,  "*.runtimeconfig.dev.json" can be replaced with variable
```

namespace Microsoft.DotNet.Cli.Utils
{
    public static class FileNameSuffixes
    {
        public const string DepsJson = ".deps.json";
        public const string RuntimeConfigJson = ".runtimeconfig.json";
        public const string RuntimeConfigDevJson = ".runtimeconfig.dev.json";
```

...
But it requires nuget package Microsoft.DotNet.Cli.Utils

Reply:
Interesting, I just cheked if that package have something else we need. But I didn't find anything else besides those consts. Let's not import it for now.